tests/benchmarks/data_structure_perf/dlist_perf/src/dlist_perf.c

```diff
@@ -173,12 +173,89 @@ void test_dlist_for_each(void)
 	zassert_true(count == 2, "SYS_DLIST_ITERATE_FROM_NODE failed");
 }
 
+/**
+ * @brief Test that access the 'head' and 'tail' in constant time
+ *
+ * @details defined a list and append several nodes.
+ * defined two pointers---'head','tail'.No matter how many nodes
+ * dlist have, get head and tail from the dlist directly.the time
+ * complexity of accessing head and tail is O(1).
+ *
+ * @ingroup lib_dlist_tests
+ */
+void test_dlist_peak_head_tail(void)
+{
+	sys_dlist_t list;
+	sys_dnode_t node[10] = {NULL};
+
+	sys_dlist_init(&list);
+
+	for (int i = 0; i < ARRAY_SIZE(node); i++) {
+		sys_dlist_append(&list, &node[i]);
+	}
+
+	sys_dlist_t *head = NULL, *tail = NULL;
+
+	/* Get 'head' node directly, the time complexity is O(1) */
+	head = list.head;
+	zassert_true(head == &node[0],
+					"dlist can't access 'head' in constant time");
+
+	/* Get 'tail' node directly, the time complexity is O(1) */
+	tail = list.tail;
+	zassert_true(tail == &node[ARRAY_SIZE(node) - 1],
+					"dlist can't access 'tail' in constant time");
+}
+
+/**
+ * @brief Test that insert or remove operates in constant time
+ *
+ * @details defined a list and append several nodes.
+ * And define a node to be ready for inserting or removing,
+ * 'insert' and 'remove' are the operations with fixed steps and
+ * no matter the size of a dlist.
+ * So verify that the operations are running in constant time.
+ *
+ * @ingroup lib_dlist_tests
+ */
+void test_dlist_insert_and_remove(void)
+{
+	sys_dlist_t list;
+	sys_dlist_t node[10] = {NULL};
+
+	sys_dlist_init(&list);
+
+	for (int i = 0; i < ARRAY_SIZE(node); i++) {
+		sys_dlist_append(&list, &node[i]);
+	}
+
+	sys_dnode_t node_ins, *insert_at = &node[ARRAY_SIZE(node)/2];
+	sys_dnode_t *insert_node = &node_ins;
+
+	/* Insert a node and steps are fixed,the time complexity is O(1) */
+	insert_node->prev = insert_at->prev;
+	insert_node->next = insert_at;
+	insert_at->prev->next = insert_node;
+	insert_at->prev = insert_node;
```

Comment:
This shows that `sys_dlist_t` is a double linked list, just wondering whether or not we should test `sys_dlist_insert` that is exactly what this code does. But I don't know how to test that function does it in O(1) as well without instrumentation.